app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -249,93 +258,68 @@ class FormController(
-        frontendAppConfig)
-    ).pure[Future]
-  }
+            suppressErrors,
+            cache,
+            processDataService.recalculateDataAndSections,
+            fileUploadService.getEnvelope,
+            validationService.validateFormComponents,
+            validationService.evaluateValidation
+          )
+          .map(handlerResult =>
+            Ok(renderer.renderSection(
+              maybeAccessCode,
+              cache.form,
+              sectionNumber,
+              handlerResult.data,
+              cache.formTemplate,
+              handlerResult.result,
+              handlerResult.envelope,
+              cache.form.envelopeId,
+              handlerResult.validatedType,
+              handlerResult.sections,
+              formMaxAttachmentSizeMB,
+              contentTypes,
+              cache.retrievals,
+              cache.form.visitsIndex.visit(sectionNumber),
+              cache.form.thirdPartyData.obligations
+            )))
+    }
 
   val choice: data.Form[String] = play.api.data.Form(
     play.api.data.Forms.single(
       "decision" -> play.api.data.Forms.nonEmptyText
     ))
 
   def decision(formTemplateId: FormTemplateId, maybeAccessCode: Option[AccessCode]): Action[AnyContent] =
-    auth.async(formTemplateId, maybeAccessCode) { implicit request => implicit l => cache =>
-      choice.bindFromRequest
-        .fold(
-          _ =>
-            BadRequest(
-              continue_form_page(
-                cache.formTemplate,
-                choice.bindFromRequest().withError("decision", "error.required"),
-                maybeAccessCode,
-                frontendAppConfig)).pure[Future], {
-            case "continue" =>
-              val dataRaw = FormDataHelpers.formDataMap(cache.form.formData) + cache.form.visitsIndex.toVisitsTuple
-              redirectWithRecalculation(cache, dataRaw, maybeAccessCode, ForceReload)
-            case "delete" =>
-              deleteForm(maybeAccessCode, cache)
-            case _ => Redirect(routes.FormController.newForm(formTemplateId)).pure[Future]
-          }
-        )
+    auth.authAndRetrieveForm(formTemplateId, maybeAccessCode, OperationWithForm.EditForm) {
+      implicit request => implicit l => cache =>
+        choice.bindFromRequest
+          .fold(
+            _ =>
+              BadRequest(
+                continue_form_page(
+                  cache.formTemplate,
+                  choice.bindFromRequest().withError("decision", "error.required"),
+                  maybeAccessCode,
+                  frontendAppConfig)).pure[Future], {
+              case "continue" =>
+                val dataRaw = FormDataHelpers.formDataMap(cache.form.formData) + cache.form.visitsIndex.toVisitsTuple
+                redirectWithRecalculation(cache, dataRaw, maybeAccessCode, ForceReload)
+              case "delete" =>
+                // ToDo: Lance - could we make this redirect to delete(., .) to make use of the permissions checking there?
```

Comment:
I guess this comment is not needed anymore, now that delete require the same permission?

Reply:
Removed